### ISSUE TITLE
Paging output of running commands; I/O Error

### ISSUE THREAD

**Issue by adi-g15** [I60R/page#23]
When piping output of an actively running command, when we have to exit, it causes a panic (after nvim has exited), with the message:

```
thread 'main' panicked at 'Read from stdin failed unexpectedly: Os { code: 5, kind: Uncategorized, message: "Input/output error" }', src/page.rs:325:75
```

I can't say if its good use case, but for eg. piping output of a make command, etc.

### Reproducing the Issue

![2022-01-09_15-53](https://user-images.githubusercontent.com/37269665/148678600-efde2721-935b-42f3-8330-bb571f8a686c.png)

![2022-01-09_16-03](https://user-images.githubusercontent.com/37269665/148678596-b740dcd1-adc7-450b-b93c-bbb00e7dae18.png)

> The make errors are due to make being killed before completing, not related to page

### Probable Solution

As of now, I can think of these

1. Try to kill the command being piped, eg. when user enters `:q` in nvim, less does that by default when it's closed as it seems

> but problem is what if the piped command has higher authority for eg. `sudo dmesg | page`

2. Dont read after nvim has exited

> Or when the piped command ends (if we can get it, i tried looking at code, dont know)

> Also, I wanted to symlink less to page, any other problematic case 😅 ?

**Comment by I60R** [comment]
Should be fixed in 3.0 version which I've recently published

> Dont read after nvim has exited

That's the approach I've took

> Also, I wanted to symlink less to page, any other problematic case ?

Personally I have it, but I don't know it it should be recommended for someone else.

Perhaps a better would be to create a different alias e.g. `pg` or something similar.

**Comment by I60R** [comment]
There's still a chance that the problem will persist, so feel free to reopen the issue (I don't promise to fix it immediately but somewhere in the future I'll definitely find time for that)
